server/src/main/java/io/unitycatalog/server/handler/CoordinatedCommitsHandler.java

```diff
@@ -0,0 +1,140 @@
+package io.unitycatalog.server.handler;
+
+import io.unitycatalog.server.model.Commit;
+import io.unitycatalog.server.model.DataSourceFormat;
+import io.unitycatalog.server.model.TableInfo;
+import io.unitycatalog.server.persist.CommitRepository;
+import io.unitycatalog.server.persist.TableRepository;
+import io.unitycatalog.server.persist.dao.CommitDAO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import static io.unitycatalog.server.persist.CommitRepository.MAX_NUM_COMMITS_PER_TABLE;
+
+public class CoordinatedCommitsHandler {
+  private static final Logger LOGGER = LoggerFactory.getLogger(CoordinatedCommitsHandler.class);
+  public static final TableRepository TABLE_REPOSITORY = TableRepository.getInstance();
+  public static final CommitRepository COMMIT_REPOSITORY = CommitRepository.getInstance();
+
+  public static void validate(Commit commit) {
+    // Validate the commit object
+    assert commit.getTableId() != null;
+    //    assert commit.getTableUri() != null;
+
+    // TODO: Add other assertions like the table URI path exists
+
+    // Validate the commit info object
+    if (commit.getCommitInfo() != null) {
+      assert commit.getCommitInfo().getIsDisownCommit() != null;
+      assert commit.getCommitInfo().getFileSize() > 0;
+      assert commit.getCommitInfo().getFileName() != null;
+      assert commit.getCommitInfo().getVersion() != null;
+      assert commit.getCommitInfo().getTimestamp() != null;
+      assert commit.getCommitInfo().getFileModificationTimestamp() != null;
+    } else {
+      // If commit info is null, then it should be a backfill only commit
+      assert commit.getLatestBackfilledVersion() != null;
+    }
+  }
+
+  public static void validateTablePath(Commit commit) {
+    TableInfo tableInfo = TABLE_REPOSITORY.getTableById(commit.getTableId());
+    //    assert tableInfo.getTableType() == TableType.MANAGED;
```

Comment:
Why do we have it commented out?

Reply:
This is because we cannot create MANAGED tables yet, so the tests in the next PR actually create a external table and test commits on that external table.